### PR TITLE
Name based cache/core management

### PR DESCRIPTION
**Pull request by robertbaldyga**

### CODE REVIEW

**Comment by KamilLepek** [comment]
retest this please

File: inc/ocf_mngt.h [hunk 1]
```diff
@@ -101,6 +101,22 @@ uint32_t ocf_mngt_cache_get_count(ocf_ctx_t ctx);
  */
 int ocf_mngt_cache_get_by_id(ocf_ctx_t ctx, ocf_cache_id_t id, ocf_cache_t *cache);
 
+/**
+ * @brief Get OCF cache by name
+ *
+ * @note This function on success also increasing reference counter
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply inc/ocf_mngt.h]: Done.

File: src/mngt/ocf_mngt_core.c [hunk 2]
```diff
@@ -21,6 +21,12 @@ static ocf_seq_no_t _ocf_mngt_get_core_seq_no(ocf_cache_t cache)
 	return ++cache->conf_meta->curr_core_seq_no;
 }
 
+static int ocf_mngt_core_set_name(ocf_core_t core, const char *name)
+{
+	return env_strncpy(core->conf_meta->name, OCF_CORE_NAME_SIZE - 1,
+			name, env_strnlen(name, OCF_CORE_NAME_SIZE));
```
Comment (arutk) [inline src/mngt/ocf_mngt_core.c]: Same here, looks like env_strnlen() doesn't add anything on top of OCF_CORE_NAME_SIZE

File: inc/ocf_mngt.h [hunk 3]
```diff
@@ -309,13 +324,15 @@ struct ocf_mngt_cache_config {
 /**
  * @brief Initialize core config to default values
  *
+ * @note This function doesn't initiialize name field which has no default
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply inc/ocf_mngt.h]: Done.

File: inc/ocf_mngt.h [hunk 4]
```diff
@@ -58,7 +58,7 @@ struct ocf_mngt_core_config {
 /**
  * @brief Initialize core config to default values
  *
- * @note This function doesn't initiialize uuid and volume_type fields
+ * @note This function doesn't initiialize name, uuid and volume_type fields
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply inc/ocf_mngt.h]: Done.

File: src/mngt/ocf_mngt_core.c [hunk 5]
```diff
@@ -593,10 +590,23 @@ void ocf_mngt_cache_add_core(ocf_cache_t cache,
...
 	context->cfg = *cfg;
 
+	name = env_vmalloc(OCF_CORE_NAME_SIZE);
+	if (!name) {
+		result = -OCF_ERR_NO_MEM;
+		goto err_pipeline;
+	}
+
+	result = env_strncpy(name, OCF_CORE_NAME_SIZE,
+			cfg->name, env_strnlen(cfg->name, OCF_CORE_NAME_SIZE));
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply src/mngt/ocf_mngt_core.c]: Done.

File: src/mngt/ocf_mngt_core.c (hunk 2 above)
Comment (robertbaldyga) [reply src/mngt/ocf_mngt_core.c]: Done.

Comment (arutk) [reply src/mngt/ocf_mngt_core.c]: I meant "OCF_CORE_NAME_SIZE - 1" , not "OCF_CORE_NAME_SIZE"

Comment (robertbaldyga) [reply src/mngt/ocf_mngt_core.c]: As it's guaranted that `env_strncpy()` always appends null terminator, changed both source and destination lengths to `OCF_CORE_NAME_SIZE`.

File: src/mngt/ocf_mngt_cache.c [hunk 6]
```diff
@@ -726,18 +726,20 @@ static int _ocf_mngt_init_prepare_cache(struct ocf_cache_mngt_init_params *param
...
+	/* Check if cache with specified name exists */
+	ret = ocf_mngt_cache_get_by_name(param->ctx, cfg->name, &cache);
+	if (!ret) {
+		ocf_mngt_cache_put(cache);
+		/* Cache already exist */
+		ret = -OCF_ERR_CACHE_EXIST;
+		goto out;
 	}
 
+	ret = env_strncpy(cache_name, sizeof(cache_name) - 1,
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply src/mngt/ocf_mngt_cache.c]: Done.

File: src/ocf_cache.c [hunk 7]
```diff
@@ -17,22 +17,17 @@ ocf_volume_t ocf_cache_get_volume(ocf_cache_t cache)
...
-{
-	OCF_CHECK_NULL(cache);
-	return cache->cache_id;
-}
-
 int ocf_cache_set_name(ocf_cache_t cache, const char *src, size_t src_size)
 {
 	OCF_CHECK_NULL(cache);
-	return env_strncpy(cache->name, OCF_CACHE_NAME_SIZE - 1, src, src_size);
+	return env_strncpy(cache->conf_meta->name, OCF_CACHE_NAME_SIZE - 1,
```
Comment (arutk) [inline src/ocf_cache.c]: "-1" not needed

Comment (robertbaldyga) [reply src/ocf_cache.c]: Done.

**Pull request merged**